### PR TITLE
feat(web-analytics): Filter test users from web analytics

### PR DESCRIPTION
**Pull request by robbie-c**
## Problem
https://posthoghelp.zendesk.com/agent/tickets/7327 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/10656)

## Changes

Filter test users from web analytics queries by borrowing some code from the hogql lifecycle code that does this, an applying it to the event/session WHERE clauses

## How did you test this code?

Ran manually

### CODE REVIEW

File: posthog/hogql_queries/web_analytics/web_analytics_query_runner.py [hunk 1]
```diff
@@ -91,17 +95,29 @@ def session_having(self, include_previous_period: Optional[bool] = None):
...
             properties,
             self.team,
         )
 
+    @cached_property
+    def _test_account_filters(self):
+        if isinstance(self.team.test_account_filters, list) and len(self.team.test_account_filters) > 0:
+            return self.team.test_account_filters
+        else:
+            return []
```
Comment (Gilbert09) [inline posthog/hogql_queries/web_analytics/web_analytics_query_runner.py:119]: I'm semi-surprised if this works considering I had to add the same functionality as:
```python
        # Filter Test Accounts
        if (
            self.query.filterTestAccounts
            and isinstance(self.team.test_account_filters, list)
            and len(self.team.test_account_filters) > 0
        ):
            for property in self.team.test_account_filters:
                filters.append(property_to_expr(property, self.team))
```

Comment (robbie-c) [reply posthog/hogql_queries/web_analytics/web_analytics_query_runner.py:119]: I based this on that code :D but it should be possible to add two python arrays together if that's what you mean? Or what was it that didn't work.

Comment (Gilbert09) [reply posthog/hogql_queries/web_analytics/web_analytics_query_runner.py:119]: Looks like you're not doing `property_to_expr` from what I could see (unless im being blind?)

Comment (robbie-c) [reply posthog/hogql_queries/web_analytics/web_analytics_query_runner.py:119]: Ah right, this happens where it's called in events_where or sessions_where

**Review by Gilbert09 — APPROVED** [review]

**Pull request merged**